### PR TITLE
Incorrect assert,  Compiler warning fix

### PR DESCRIPTION
**Pull request by DeonMarais64**
Hi,

Please check proposed changes.

Regards,
Deon Marais

### CODE REVIEW

File: src/lv_misc/lv_color.h [hunk 1]
```diff
@@ -373,9 +373,10 @@ static inline uint16_t lv_color_to16(lv_color_t color)
 #endif
     LV_COLOR_SET_B16(ret, LV_COLOR_GET_B(color) >> 3);  /* 8 - 5  = 3*/
     return ret.full;
-#endif
-
+#else
```
Comment (embeddedt) [inline src/lv_misc/lv_color.h]: I don't think it should be possible to reach this `else` clause if `lv_conf.h` is configured correctly. What is the value of `LV_COLOR_DEPTH` in your case?

Comment (DeonMarais64) [reply src/lv_misc/lv_color.h]: 16 bits, 
Using the IAR compiler gives 
Warning[Pe111]: statement is unreachable 
Which it is .. :)

Comment (embeddedt) [reply src/lv_misc/lv_color.h]: In that case why wouldn't it use [this clause](https://github.com/littlevgl/lvgl/blob/1765dde018cf6d084d3a48371eca54e1bdfb668f/src/lv_misc/lv_color.h#L363)?

Comment (DeonMarais64) [reply src/lv_misc/lv_color.h]: On reflection a better solution IMO would be to remove the final return altogether,
https://github.com/littlevgl/lvgl/blob/10b9c9b2f5344e7b2f5cc00a19ed86ed56ae9866/src/lv_misc/lv_color.h#L378
that would make it consistent with the other ``lv_color_to??`` functions.

Comment (kisvegabor) [reply src/lv_misc/lv_color.h]: > On reflection a better solution IMO would be to remove the final return altogether,

I agree.

Comment (embeddedt) [reply src/lv_misc/lv_color.h]: It turns out that I completely misunderstood the issue; sorry about that. :slightly_frowning_face:

The latest changes are fine with me.

**Comment by kisvegabor** [comment]
Thank you!

**Pull request merged**